manifests/init.pp

```diff
@@ -148,6 +150,7 @@
   Boolean $manage_service                        = true,
   Boolean $manage_service_file                   = false,
   Boolean $restart_on_change                     = true,
+  Array[String[3]] $env_vars                     = [],
```

Comment:
Why 3?

Reply:
shortest possible env var would be something like `A=B`. I think a lower string length isn't possible.

Reply:
Empty value to unset something? Also wondering if you want to make it a hash instead.

Reply:
```puppet
  Optional[Array[String[3]]] $env_vars = [],
```

Might be more clear here.

Reply:
That doesn't make sense. It's equivalent to:
```puppet
Variant[Undef, Array[String[3]]] $env_vars = [],
```

However, if you "call" it as such:
```puppet
class { 'nomad':
  env_vars => undef,
}
```

Then Puppet resolves it to what the default is, which is `[]`. Thus there is no way to pass in `undef` and `Optional` is just redundant.

Reply:
I updated it to work with a hash. can you please take a look again?